components/dropdown/dropdown-content-mixin.js

```diff
@@ -1001,6 +1009,14 @@ export const DropdownContentMixin = superclass => class extends LocalizeCoreElem
 				return spaceAround.right * -1;
 			}
 		}
+		if (this.align === 'start' || this.align === 'end') {
+			const shift = Math.min((targetWidth / 2) - (20 + 16 / 2), 0);
+			if (this.align === 'start') {
+				return shift;
+			} else {
+				return targetWidth - contentWidth - shift;
+			}
+		}
```

Comment:
This is at the bottom because the above cases still apply for the cases where there isn't enough space to align right at the end or at the start.